docs/_source/docs/resolvers.rst

```diff
@@ -52,6 +52,14 @@ file_contents
 
 **deprecated**: Consider using the `file`_ resolver instead.
 
+no_value
+~~~~~~~~
+
+This resolver "resolves to nothing", functioning just as if it was not set at all. This works just
+like the "AWS::NoValue" special variable that you can reference on a CloudFormation template. It
+can help simplify Stack and StackGroup config Jinja logic in cases where, if a condition is met, a
+value is passed, otherwise no value is passed.
```

Comment:
I'm having a difficult time understand the general use case for this resolver.   Is it something that's only used for debugging purposes?  I just can't figure out when I would ever set this in my sceptre files.  Can you please help me out?

Reply:
So there's two things going on here:
1. Resolvers that resolve to `None` are removed from their container. 
2. `!no_value` is a resolver that exclusively resolves to `None` without any other functionality. 

It works just like `AWS::NoValue`, which is pretty useful in CloudFormation templates where you have logic like:
`!If [ 'MyCondition', !Ref MyVPCId, !Ref AWS::NoValue ]`. In a CloudFormation template, that condition would result in no value being set on that key if 'MyCondition' was equal to False.

In the context of Sceptre, imagine this scenario:

```yaml
parameters:
    my_parameter: {{ var.get('some_value_that_might_not_be_set', '!no_value' }}
```

Imagine the CloudFormation template has a default value for my_parameter. But if you pass any string at all for that parameter, that default value will not actually be used and it will just use the parameter value you've passed. But by using `!no_value` in this way, we won't actually pass "my_parameter" at all to the template if "some_value_that_might_not_be_set" is not set on the `var`.

Could you accomplish the same via other means in Jinja? Sure. Would it be more elegant? It depends on the way the user is setting up their Stack and StackGroups. `!no_value` gives the user an option that they can use. It gets more important when you talk about inheriting things from StackGroups.

Reply:
BINGO! your example made it crystal clear to me, thanks.  Can we incorporate an example like that into the docs?  

also, would this be an equivalent example?
```
parameters:
    my_parameter: {{ environment_variable.MY_VAR  |  no_value }}  
```

Reply:
> Can we incorporate an example like that into the docs?

Yeah, not a problem. 

> would this be an equivalent example?

Almost, but not quite. Remember, it's a resolver, so it would need to be:
```yaml
parameters:
    my_parameter: {{ environment_variable.MY_VAR  |  "!no_value" }} 
```